deCODE/deCODE_by_CHR.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+# Import library for analysis-runner
+import click
+import hail as hl
+from cpg_utils.hail_batch import output_path
+
+# Import library for deCODE job
+import pandas as pd
+from pandas import Series,DataFrame
+import numpy as np
+import math
+import collections
+
+@click.command()
+@click.option('--dataset', help="data to query")
+@click.option('--chrom', help="chromsome")
+@click.option('--output', help="output name")
+@click.option('--rerun', is_flag=True, help='Whether to overwrite cached files', default=False)
+def query(dataset, chrom, output, rerun):
+    p_out = output_path(output)
+    log_out = output_path(output + ".log")
+
+    hl.init(default_reference='GRCh38')
+    hl.context.warning("inited")
+    hl.copy_log(log_out)
+
+    if rerun or not hl.hadoop_exists(p_out):
+        # Section 1 - Read dataset
+        mt = hl.read_matrix_table(dataset)
+        mt = mt.filter_rows(mt.locus.contig == chrom) 
```

Comment:
Rather than two separate files (one for running on everything, and one for running per chromosome), you could just make --chrom an optional parameter so that you only need to maintain one script.

If you give `chrom` a default value of `None` on line 17, then you can put an `if` around this line to only filter based on `chrom` if it is not `None`. That way you run on all the data if no `--chrom` is provided, or the provided chromosome(s) if this option is passed.

Reply:
Sorry again for the confusion! I have deleted deCODE.py (as it is not being used).